src/components/forms/NetworkDevicesForm/NetworkDevice.tsx

```diff
@@ -1,129 +1,57 @@
 import { type FC } from "react";
-import type { LxdNicDevice } from "types/device";
+import type { LxdNicDevice, LxdNoneDevice } from "types/device";
 import type { LxdNetwork } from "types/network";
 import type { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
-import NetworkSelector from "pages/projects/forms/NetworkSelector";
-import { Button, Icon } from "@canonical/react-components";
-import { ensureEditMode } from "util/instanceEdit";
-import NetworkDeviceAcls from "components/forms/NetworkDevicesForm/NetworkDeviceAcls";
-import ResourceLink from "components/ResourceLink";
-import { supportsNicDeviceAcls } from "util/networks";
 import { useNetworks } from "context/useNetworks";
+import NetworkDeviceActionButtons from "./NetworkDeviceActionButtons";
+import NetworkDeviceContent from "./NetworkDeviceContent";
+import { isNicDevice } from "util/devices";
+import type { InheritedNetwork } from "util/configInheritance";
 
 interface Props {
   index: number;
   readOnly: boolean;
   project: string;
   formik: InstanceAndProfileFormikProps;
-  focusNetwork: (id: number) => void;
-  removeNetwork: (id: number) => void;
-  device: LxdNicDevice;
+  device?: LxdNicDevice | LxdNoneDevice;
   network?: LxdNetwork;
+  inheritedDevice?: InheritedNetwork;
 }
 
 const NetworkDevice: FC<Props> = ({
   index,
```

Comment:
Can we remove the index here and in the children? We already pass the device, and I think in the nested children we can use the `getIndex` function instead of prop drilling to simplify the interfaces in between.
```suggestion
```